test/micro/org/openjdk/bench/java/lang/foreign/AllocFromTest.java

```diff
@@ -0,0 +1,178 @@
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.openjdk.bench.java.lang.foreign;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.lang.foreign.Arena;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.MemorySegment.Scope;
+import java.lang.foreign.SegmentAllocator;
+import java.lang.foreign.ValueLayout;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@State(org.openjdk.jmh.annotations.Scope.Thread)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Fork(value = 3, jvmArgsAppend = { "--enable-native-access=ALL-UNNAMED" })
+public class AllocFromTest extends CLayouts {
+
+    Arena arena = Arena.ofConfined();
+
+    SlicingPool pool = new SlicingPool();
+
+    @Param({"5", "20", "100", "500", "1000"})
+    public int size;
+    public byte[] arr;
+
+    @Setup
+    public void setup() {
+        arr = new byte[size];
+        Random random = new Random();
+        random.nextBytes(arr);
+    }
+
+    @Benchmark
+    public MemorySegment alloc() {
+        Arena arena = Arena.ofConfined();
+        MemorySegment segment = arena.allocateFrom(ValueLayout.JAVA_BYTE, arr);
+        arena.close();
+        return segment;
+    }
+
+    @Benchmark
+    public long alloc_malloc_arena() {
+        MallocArena arena = new MallocArena();
+        MemorySegment segment = arena.allocateFrom(ValueLayout.JAVA_BYTE, arr);
+        arena.close();
+        return segment.address();
```

Comment:
Why not return the `segment` here as well? (like in `alloc`). Seems like an unfair comparison?